lib/src/exceptions.dart

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+class LoggingException implements Exception {
+  const LoggingException();
+}
+
+class HttpException implements IOException {
+  final String message;
+  final Uri? uri;
+
+  const HttpException(this.message, {this.uri});
+
+  @override
+  String toString() {
+    final b = StringBuffer()..write('HttpException: ')..write(message);
+    final uri = this.uri;
+    if (uri != null) {
+      b.write(', uri = $uri');
+    }
+    return b.toString();
+  }
+}
+
+class CronetException implements Exception {
+  final int val;
```

Comment:
For `HttpException` we are getting the error string from a error pointer provided by cronet to `onFailed` callback. Other than that, cronet just declares enums like -
```
Cronet_RESULT_SUCCESS = 0,
Cronet_RESULT_ILLEGAL_ARGUMENT = -100,
Cronet_RESULT_ILLEGAL_ARGUMENT_STORAGE_PATH_MUST_EXIST = -101,
```
and so on... I can't find any proper function that takes this type of enum and gives us a string. Can we generate the string on our own if we have the int value?

Reply:
Does ffigen support getting the strings of the enums? Could it?

If ffigen cannot we should probably just move them over to Dart code so that we can give a bit more proper error messages to users.

Reply:
> Does ffigen support getting the strings of the enums?

Nope.

Reply:
> If ffigen cannot we should probably just move them over to Dart code so that we can give a bit more proper error messages to users.

Okay.